posthog/migrations/0803_dwh_source_refactor.py

```diff
@@ -0,0 +1,152 @@
+# Generated by Django 4.2.22 on 2025-07-21 11:26
+
+from django.db import migrations
+from posthog.warehouse.models import ExternalDataSource as ExternalDataSourceModel
+
+
+def forwards(apps, schema_editor):
+    ExternalDataSource: ExternalDataSourceModel = apps.get_model("posthog", "ExternalDataSource")
+    all_sources = ExternalDataSource.objects.all()
+
+    for source in all_sources:
+        source_type = ExternalDataSourceModel.Type(source.source_type)
```

Comment:
Thoughts on making a brand new data model and cloning over the existing sources entirely with new fields or something related that would allows us to jump back? This migration seems quite high risk and unrevertable. It'd be a struggle if we dropped keys